AK/Time.h

```diff
@@ -44,9 +44,31 @@ inline void timeval_add(const TimevalType& a, const TimevalType& b, TimevalType&
 {
     result.tv_sec = a.tv_sec + b.tv_sec;
     result.tv_usec = a.tv_usec + b.tv_usec;
-    if (result.tv_usec > 1000000) {
+    if (result.tv_usec > 1000 * 1000) {
         ++result.tv_sec;
-        result.tv_usec -= 1000000;
+        result.tv_usec -= 1000 * 1000;
+    }
+}
+
+template<typename TimespecType>
+inline void timespec_sub(const TimespecType& a, const TimespecType& b, TimespecType& result)
+{
+    result.tv_sec = a.tv_sec - b.tv_sec;
+    result.tv_nsec = a.tv_nsec - b.tv_nsec;
+    if (result.tv_nsec < 0) {
+        --result.tv_sec;
+        result.tv_nsec += 1000 * 1000 * 1000;
```

Comment:
*Perhaps* it'd be better to write it using C++14 ticks as `1'000'000'000`? Not sure it'd be better though.

Reply:
Ah yeah, I did that in sys/time.h but then figured it doesn't work in C. It would work in this file, but then it'd be different from sys/time.h, which seems fine actually. Changing, thanks.